Content.Server/Antag/AntagRandomObjectivesSystem.cs

```diff
@@ -32,17 +33,24 @@ private void OnAntagSelected(Entity<AntagRandomObjectivesComponent> ent, ref Aft
         }
 
         var difficulty = 0f;
-        foreach (var set in ent.Comp.Sets)
+        var assignedTargetObjectives = new HashSet<EntityUid?>();
+        foreach (var set in ent.Comp.Sets.Where(set => _random.Prob(set.Prob)))
         {
-            if (!_random.Prob(set.Prob))
-                continue;
-
             for (var pick = 0; pick < set.MaxPicks && ent.Comp.MaxDifficulty > difficulty; pick++)
             {
                 var remainingDifficulty = ent.Comp.MaxDifficulty - difficulty;
                 if (_objectives.GetRandomObjective(mindId, mind, set.Groups, remainingDifficulty) is not { } objective)
                     continue;
 
+                if (TryComp<TargetObjectiveComponent>(objective, out var targetObjective))
```

Comment:
I'm not commenting on if this is the right solution (Too tired right now 😆) but make sure to add some documentation in the target objective saying two target objectives can't target the same thing.

This is (probably) going to ruin objectives like mine (#30360) which is fine but without a comment It is basically impossible to know target objectives have this property. It also is going to force one kill objective per player total. I that's probably fine but again, its probably a good idea to add documentation about it!